src/mail_room.py

```diff
@@ -0,0 +1,166 @@
+    """Return the user response to a prompt."""
+    try:
+        input = raw_input
+    except NameError:
+        pass
+    usr_answer = input('{} \n>>>>>> '.format(prompt))
+    return usr_answer.lower()
+
+
+def validator(usr_response):
+    """Check the user answer."""
+    usr_response = usr_response.lower()
+    if usr_response == "send a thank you":
+        return True
+    elif usr_response == "create a report":
+        return False
+    elif usr_response == "q":
+        sys.exit()
+    else:
+        return "error"
+
+
+def send_or_report():
+    """Send a thank you or create a report."""
+    usr_response = prompt_usr(prompts[0])
+    validate = validator(usr_response)
+    if validate is True:
+        return True
+    elif validate is False:
+        return False
+    elif validate == "error":
+        print("Please choose a valid option")
+        send_or_report()
+
+
+def get_name():
+    """Get gifter name from user."""
+    while True:
+        usr_response = prompt_usr(prompts[1])
+        usr_response = usr_response.lower()
+        if usr_response == "q":
+            sys.exit()
+        elif usr_response == "l":
+            name_list = create_list(GIFTER_DICT)
+            print(name_list)
+        else:
+            break
+    return usr_response
+
+
+def check_dict(name, donate_dict):
+    """Check to see if name exists in dictionary."""
+    if name in donate_dict:
+        return True
+    else:
+        donate_dict.update({str(name): {'name': str(name), 'total_ammt': 0, 'most_recent_ammt': 0, 'times_given': 0}})
+        return False
+
+
+def get_ammt():
+    """Get donation ammount from user."""
+    while True:
+        try:
+            usr_response = prompt_usr(prompts[2])
+            usr_response = usr_response.lower()
+            if usr_response == "q":
+                sys.exit()
+            return int(usr_response)
+        except ValueError:
+            print("Please type a full dollar ammount with no special characters")
+
+
+def update_dict(name, ammt):
+    """Update values for doner in dict."""
+    temp_key = GIFTER_DICT.get(name)
+    get_total = temp_key.get('total_ammt')
+    total = int(get_total) + ammt
+    get_total_times = temp_key.get('times_given')
+    total_times = (get_total_times + 1)
+    GIFTER_DICT.update({name: {'name': name, 'total_ammt': total, 'most_recent_ammt': ammt, 'times_given': total_times}})
+
+
+def send_email(name, ammt):
+    """Print out customized email text."""
+    print(email_text.format(name, ammt))
+    return email_text.format(name, ammt)
+
+
+def create_list(donate_dict):
+    """Create a list of names that have given before."""
+    if len(donate_dict) > 0:
+        name_list = list(donate_dict.keys())
+        return name_list
+    else:
+        print("There are no names")
+
+
+def create_report():
+    """Create a report of donaters."""
+    if len(GIFTER_DICT) == 0:
+        print("there is nothing to report")
+        return False
```

Comment:
This is a prime example of asking permission instead of forgiveness.  As it turns out there is no actual purpose to the first three lines of this function.  Let's talk it through:

If the GIFTER_DICT has no entries in it, then what is the result of calling `create_list` and passing it in on line 138 below?  If you modify the function as I suggested above, then there's no reason at all to check the length of this dict.  If there are no entries, then `name_list` will be an empty list.  Iterating over an empty list does nothing, and `gifter_list` will be an empty list.  You could return that empty list and allow the calling context to decide what to do with it.  

This would help to isolate the places where you interact with the user by printing or reading values to the top-level main function.  That's a good thing.